TitanRobotics2024/src/main/java/frc/robot/Subsystem/Gyro.java

```diff
@@ -0,0 +1,99 @@
+package frc.robot.Subsystem;
+
+import com.kauailabs.navx.frc.AHRS; //https://dev.studica.com/releases/2024/NavX.json
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Gyro implements Subsystem
+{
+
+  private static Gyro instance = null;
+  AHRS ahrs;
+  SmartDashboardSubsystem smartDashboardSubsystem;
+
+  double yAxisRate;
+  double xAxisRate;
+  double pitchAngleDegrees = 0;
+  double rollAngleDegrees = 0;
+  double yawAngleDegrees = 0;
+  double angleDegrees = 0;
+
+  public static Gyro getInstance()
+  {
+    if (instance == null)
+    {
+      instance = new Gyro();
+    }
+    return instance;
+  }
+
+  public Gyro()
+  {
+    smartDashboardSubsystem = SmartDashboardSubsystem.getInstance();
+    try
+    {
+      ahrs = new AHRS(SPI.Port.kMXP);
+    }
+    catch (Exception e)
+    {
+      System.out.println(
```

Comment:
Remove system.out.println statements

Reply:
useful for error checking, will remove when smartDashboardSubsystem.error works properly

Reply:
This will actually lead to decreased robot performance, remove. You will be spamming the rio with printing messages instead of letting the rio run the robot